functions/agencies/__TESTS__/handler.test.ts

```diff
@@ -1,35 +1,135 @@
-  awsRequestId: 'mocked',
-  logGroupName: 'mocked',
-  logStreamName: 'mocked',
-  getRemainingTimeInMillis(): number {
-    return 999;
-  },
-  done(error?: Error, result?: any): void {
-    return;
-  },
-  fail(error: Error | string): void {
-    return;
+const mockedEvent: Event<AgenciesRequest> = {
+  body: {
+    action: 'CREATE',
+    address: 'Some address',
+    name: 'My agency',
+    ownerId: 'owner#email',
+    phone: '1234567890'
   },
-  succeed(messageOrObject: any): void {
-    return;
-  }
+  environment: 'test'
 };
 
+const ddbMock = mockClient(DynamoDBDocumentClient);
+
+const mockDB = (tableName: string, statusCode = 200) => {
+  if (statusCode === 200) {
+    return ddbMock.on(PutCommand, { TableName: tableName }).resolves({
+      $metadata: {
+        httpStatusCode: statusCode
+      }
+    });
+  }
+
+  ddbMock.on(PutCommand, { TableName: tableName }).rejects();
+}
+
 describe('agencies handler', () => {
-  it('Should execute correctly', async () => {
-    const response = await handler({}, mockedContext, () => undefined);
-    expect(response).toStrictEqual({
-      "body": "{\"message\":\"Agencies function\"}",
-      "statusCode": 200,
+  beforeEach(() => {
+    vi.useFakeTimers();
+  });
+
+  afterEach(() => {
+    vi.useRealTimers();
+    ddbMock.reset();
+  });
+
+  describe('Success', () => {
+    it('Should execute creation correctly', async () => {
+      const date = new Date(2000, 1, 1, 13)
+      vi.setSystemTime(date);
+
+      mockDB('agencies-test-table');
+
+      const response = await handler(mockedEvent, mockedContext, () => undefined) as Response<AgenciesResponse>;
+
+      expect(response.statusCode).toBe(StatusCodes.ok);
+      expect(response.body.agencyId).toBe(`${date.getTime()}-${mockedEvent.body.ownerId}`);
+    });
+
+    it('Should execute creation correctly when optional values are not present', async () => {
+      const date = new Date(2000, 1, 1, 13)
+      vi.setSystemTime(date);
+      const event = { ...mockedEvent, body: { ...mockedEvent.body, address: undefined, name: null, phone: null } }
+      mockDB('agencies-test-table');
+
+      //@ts-ignore
+      const response = await handler(event, mockedContext, () => undefined) as Response<AgenciesResponse>;
+
+      expect(response.statusCode).toBe(StatusCodes.ok);
+      expect(response.body.agencyId).toBe(`${date.getTime()}-${mockedEvent.body.ownerId}`);
+    });
+
+    it('Should execute creation correctly when is prod', async () => {
+      const date = new Date(2000, 1, 1, 13)
+      vi.setSystemTime(date);
+
+      // This should break given that I only mocked the test table not the prod table right?
+      mockDB('agencies-test-table');
+
+      const response = await handler({ ...mockedEvent, environment: 'prod' }, mockedContext, () => undefined) as Response<AgenciesResponse>;
+
+      expect(response.statusCode).toBe(StatusCodes.ok);
+      expect(response.body.agencyId).toBe(`${date.getTime()}-${mockedEvent.body.ownerId}`);
```

Comment:
I expect this test case to fail, given that I only mocked the table `agencies-test-table` but this is passing just fine, which makes me wonder If this is really mocking that specific table or just all interactions with the DB.